ADALiOS/ADALiOS/ADTokenCacheItem.m

```diff
@@ -169,4 +186,48 @@ - (NSString*)description
             [NSString adIsStringNilOrBlank:_refreshToken] ? @"(nil)" : @"(present)", _resource];
 }
 
+- (NSString *)clientId
+{
+    return _clientId;
+}
+
+- (void)setClientId:(NSString *)clientId
+{
+    _clientId = clientId;
+    [self calculateHash];
+}
+
+- (ADUserInformation *)userInformation
+{
+    return _userInformation;
+}
+
+- (void)setUserInformation:(ADUserInformation *)userInformation
+{
+    _userInformation = userInformation;
+    [self calculateHash];
```

Comment:
How expensive exactly? Hash is something we introduced. It is not a system property override as far as I can tell.